File: Practica 5 y 6/java/Persona.java
import java.util.*;

public class Persona {
    private String nombre, nombres, appat, apmat, ocupacion, id, nacimiento, diaN, mesN, agnoN;
    private int edad;

    //Definimos los constructores
    
    public Persona(String n, String ap, String am, String o, String dia, String mes, String agno){
        nombres = n;
        appat = ap;
        apmat = am;
        nombre = n + " " + ap + " " + am;
        diaN = dia;
        mesN = mes;
        agnoN = agno;
        ocupacion = o;
        nacimiento = dia + "/" + mes + "/" + agno;
    }
    
    public Persona(){
    }
    
    //Funciones Getters and Stters

    public String getNombre(){
        return nombre;
    }

    public String getNacimiento(){
        return nacimiento;
    }

    public String getOcupacion(){
        return ocupacion;
    }

    public void setNombre(String n, String ap, String am){
        nombres = n; appat = ap; apmat = am;
        nombre = n + " " + ap + " " + am;
    }

    public void setNacimiento(String dia, String mes, String agno){
        diaN = dia; mesN = mes; agnoN = agno;
        nacimiento = dia + "/" + mes + "/" + agno; 
    }

    public void setOcupacion(String o){
        ocupacion = o;
    }

    public String toString(){
        return "Nombre: " + getNombre() + " \nEdad: " + edad + " \nOcupacion: " + getOcupacion() 
               + "\nID: " + id + "\nNacimiento: " + getNacimiento();
    }
    
    // Calculamos información del usuario

    public void calculateEdad(){
        Calendar fechaHoy = Calendar.getInstance();
        int diaH = fechaHoy.get(Calendar.DAY_OF_MONTH);
        int mesH = fechaHoy.get(Calendar.MONTH);
        int agnoH = fechaHoy.get(Calendar.YEAR);

        int mesn = Integer.parseInt(mesN);
        int dian = Integer.parseInt(diaN);
        int agnon = Integer.parseInt(agnoN);
        
        edad = agnoH - agnon;
        if((mesH > mesn) && (diaH > dian))
            edad--;
    }

    public void calculateID(){
        String newyear = String.valueOf(agnoN);
        newyear = newyear.substring(newyear.length() - 2);
        String nom = nombres.substring(0,1);
        String apa = appat.substring(0,2).toUpperCase();
        String ama = apmat.substring(0,1);

        id = apa + ama + nom + newyear + mesN + diaN + "XXX";
    }

}

File: Practica 5 y 6/python/Persona.py
from datetime import datetime

class Persona:
    def __init__(self, n, ap, am, o, dia, mes, agno):
        self._nombres = n
        self._appat = ap
        self._apmat = am
        self._nombre = f"{n} {ap} {am}"
        self._diaN = dia
        self._mesN = mes
        self._agnoN = agno
        self._ocupacion = o
        self._nacimiento = f"{dia}/{mes}/{agno}"
        self._edad = 0
        self._id = ""

    # Funciones Getters y Setters
    def get_nombre(self):
        return self._nombre

    def get_nacimiento(self):
        return self._nacimiento

    def get_ocupacion(self):
        return self._ocupacion

    def set_nombre(self, n, ap, am):
        self._nombres = n
        self._appat = ap
        self._apmat = am
        self._nombre = f"{n} {ap} {am}"

    def set_nacimiento(self, dia, mes, agno):
        self._diaN = dia
        self._mesN = mes
        self._agnoN = agno
        self._nacimiento = f"{dia}/{mes}/{agno}"

    def set_ocupacion(self, o):
        self._ocupacion = o

    def __str__(self):
        return f"Nombre: {self.get_nombre()}\nEdad: {self._edad}\nOcupación: {self.get_ocupacion()}\nID: {self._id}\nNacimiento: {self.get_nacimiento()}"

    # Calculamos información del usuario
    def calculate_edad(self):
        fecha_hoy = datetime.now()
        diaH = fecha_hoy.day
        mesH = fecha_hoy.month
        agnoH = fecha_hoy.year

        mesn = int(self._mesN)
        dian = int(self._diaN)
        agnon = int(self._agnoN)

        self._edad = agnoH - agnon
        if mesH > mesn and diaH > dian:
            self._edad -= 1

    def calculate_id(self):
        newyear = str(self._agnoN)[-2:]
        nom = self._nombres[0]
        apa = self._appat[:2].upper()
        ama = self._apmat[0]

        self._id = f"{apa}{ama}{nom}{newyear}{self._mesN}{self._diaN}XXX"

        